Update issue templates

misherwa opened the pull request:
Adding an issue template file as mentioned in Design Document 2 that the issue template is required. 

The template includes a description, link, optical screenshots, and additional context sections.

misherwa committed:
Update issue templates

Adding an issue template file as mentioned in Design Document 2 that the issue template is required. 

The template includes a description, link, optical screenshots, and additional context sections.

misherwa committed:
Update issue_template.md

Adding title "ISSUES" and label "Issue" to the code.

thecatfour commented:
I think "Screenshots (Optional)" should be renamed to "Screenshots/Example (Optional)" to emphasize that a small example of a bug should be showcased.

misherwa committed:
Update issue_template.md

Changed "Screenshot (optional) to "Screenshots/Example (Optional)" on line 16.

As mentioned by Leslie, line 16 should be changed to  "Screenshots/Example (Optional)" to highlight a small example that should be showcased by the person creating an issue,


https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub/pull/19#issuecomment-2435890351

misherwa committed:
Small grammer fix in issue_template.md

Just removed an extra parentheses on line 16

misherwa commented:
Thanks for your input.

I made changes to line 16 by adding the "/Example" and it looks like this:
![q](https://github.com/user-attachments/assets/c44d0a99-ca51-4ae8-8bdb-89585cae848a)

barrioskevin commented:
was it possible to make the issue template on your own fork and then make the pull request? right now it looks like you did your work here on the main repository which should **not** be the case. Unless of course there is something I'm missing and you needed to work directly on this repo. 

The issue template looks good although I suggest we add a "potential solution" section or at least a section where we can write in what file or files would need to be changed in order for the fix. Also maybe like a frontend, backend, or both category

barrioskevin commented on .github/ISSUE_TEMPLATE/issue_template.md:
```diff
@@ -0,0 +1,20 @@
+
+**Description of the issue**
+Describe the issue and share important details such as the desired solution, error messages, and where the issue originates.
+
```
**Files that need to be updated** (something along those lines)
Describe what files or directories that would need to be changed / updated to fix issue.
(Im saying this should be different then the description.)

misherwa replied:
Gotcha, I'll add this

barrioskevin commented on .github/ISSUE_TEMPLATE/issue_template.md:
```diff
@@ -0,0 +1,20 @@
+---
+
+**Description of the issue**
+Describe the issue and share important details such as the desired solution, error messages, and where the issue originates.
```
Describe the issue and share important details such as domain (frontend, backend, or both), error messages, and where the issue originates or is seen. (just my suggestion not forcing you to make any changes).

misherwa replied:
I think you're right about adding details like what domain the issue is from.

misherwa commented:
> was it possible to make the issue template on your own fork and then make the pull request? right now it looks like you did your work here on the main repository which should **not** be the case. Unless of course there is something I'm missing and you needed to work directly on this repo.
> 
> The issue template looks good although I suggest we add a "potential solution" section or at least a section where we can write in what file or files would need to be changed in order for the fix. Also maybe like a frontend, backend, or both category

In our main repository, if you go to settings and scroll down to features. This is what it looks like:
![q](https://github.com/user-attachments/assets/33d5e2f3-c5fc-48fb-8500-f8b178fc6380)

If I try creating an issues template in my fork the same way by going into settings, I get this:
![q2](https://github.com/user-attachments/assets/535d51d1-bbdf-4b2f-9032-7757322d35ef)

So there isn't a "Set up templates" button when I try doing it on my fork. I could try creating a .md file on my fork and then make a PR for it, and it should work. Do you want me to try doing that?

barrioskevin commented:
I think it's kinda too late now. we should stick with how you have it now but once the PR gets closed delete the added branch and in the future we stick to editing our forks. I believe you need to enable  the feature in your fork in order to see the button or creating the md template file in the appropriate directory should also work. Sorry I can't suggest a proper solution im not all familiar with the whole template thing. In regards to the actual template I think we should include a domain section where we specify (frontend, backend, or both).

misherwa committed:
Including new requirements to issue_template.md

Added "Files that need to be updated" as a requirement in the template. 

Added detail requirements such as what domain the issue is from in the "Description of the Issue".

misherwa commented:
> I think it's kinda too late now. we should stick with how you have it now but once the PR gets closed delete the added branch and in the future we stick to editing our forks. I believe you need to enable the feature in your fork in order to see the button or creating the md template file in the appropriate directory should also work. Sorry I can't suggest a proper solution im not all familiar with the whole template thing. In regards to the actual template I think we should include a domain section where we specify (frontend, backend, or both).


No worries, I'll try to make sure I'm doing things in my fork instead of the main repository, and I'll delete the branch once this PR gets pulled.
 
I just updated the template to include the domain affected in the "Description of the issue":
![q](https://github.com/user-attachments/assets/36d3cc4e-5dbb-4e1f-8e9f-d85cb84ae354)

And the added "Files that need to be updated" requirement:
![q2](https://github.com/user-attachments/assets/3cc371c3-537b-458d-9120-af10370e8d95)

barrioskevin commented:
Looks good to me!

nnguyen144 commented:
I think this issue template will help us organize our issues and connect our pull request with them. However I don't think the 
>**Link**
>Include a link to the correlated PR number, or any resources for greater context.

is necessary (at least for the PR part) since we can already link a pull request with an issue through GitHub.

leed85 reviewed (approved):
looks good, the only thing maybe is to add the goal of what the issue was trying to fix to make it more clear for others to make suggestions
